src/App.tsx

```diff
@@ -1,61 +1,108 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { useState, FormEvent, FC } from 'react';
+import { Todo } from './types/Todo';
+
+enum DefaultField {
+  title = 'Enter a title',
+  select = 'Choose a user',
+}
+
+const initialTodos: Todo[] = todosFromServer;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    setErrorTitle(!title.trim());
+    setErrorUser(!userId);
+
+    const formattedTitle = title.trim();
+
+    if (formattedTitle.length === 0 || !userId) {
+      return;
+    }
+
+    const ids = todos.map(({ id }) => id);
+    const id = Math.max(...ids, 0) + 1;
+
+    const newTodo: Todo = {
+      id,
+      title,
+      userId,
+      completed: false,
+    };
+
+    setTodos(prev => [...prev, newTodo]);
+    reset();
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setErrorTitle(false);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setErrorUser(false);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">Title: </label>
+          <input
+            type="text"
+            value={title}
+            placeholder={DefaultField.title}
+            data-cy="titleInput"
+            onChange={handleTitleChange}
+          />
+          {errorTitle && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="userSelect">User: </label>
```

Comment:
The `label` element's `htmlFor` attribute references `userSelect`, but the `select` element does not have a corresponding `id` attribute. Add `id="userSelect"` to the `select` element to ensure the label is correctly associated with the select.